LPS-202020 fixes error of incorrect positioning of the columns visibility button

matuzalemsteles opened the pull request:
[LPS-202020](https://liferay.atlassian.net/browse/LPS-202020) Hey guys, this PR is a WIP to fix the behavior where the width of the FDS columns has some inconsistent sizes.

So far I'm still analyzing to investigate the real cause of the problem, I'm still not sure if the bug has occurred due to the changes https://github.com/liferay-platform-experience/liferay-portal/pull/228, like the changes they only affect when FF is enabled, it wouldn't make much sense, but as Markos did a test of reversing the commit and apparently that would solve it, but I'm not able to reproduce this behavior by reversing the commits and even removing the code manually in master, I still see the width with the wrong sizes.

My assumption is that as FDS takes the column size as soon as the table is rendered the first time, the sizes are already strange.

![Screenshot 2023-11-29 at 20 19 26](https://github.com/liferay-frontend/liferay-portal/assets/13750819/35351365-7811-466b-aa52-dc2c60885416)

My assumption is that this is more related to CSS because the table is already rendering with a strange size, I'm not sure if this is related to the rows having no content. I'll continue investigating.

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/fbe589eb729529318c9cc3232010549611fa1564">fbe589eb729529318c9cc3232010549611fa1564</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/matuzalemsteles/liferay-portal/tree/fix-fds">fix-fds</a><br/>Branch GIT ID: <a href="https://github.com/matuzalemsteles/liferay-portal/commit/e052c0eee1980b91c546aefd3235044de2a65eaa">e052c0eee1980b91c546aefd3235044de2a65eaa</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-36.liferay.com/job/test-portal-source-format/2299/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-36.liferay.com/userContent/jobs/test-portal-source-format/builds/2299/jenkins-report.html">here</a>.</h5></details></html>

matuzalemsteles commented:
cc @markocikos

dsanz commented:
Thanks for taking a look @matuzalemsteles . As you said, column widths are calculated and preset on a first render, and this applies even for the table headers. At a first glance I did not see any significant change on the FDS side.

I would not correlate the issue with the FF yet, because FF was added almost 3 months ago. 

A possible way to diagnose this would be to pick a table such as the client extensions list or the product table, then check what changed in the markup from a 2-3 week old commit and compare. I'd start with the stylings involved in the table when the FF is off.

matuzalemsteles commented:
> I would not correlate the issue with the FF yet, because FF was added almost 3 months ago.

Yeah the FF is not the problem here, what I want to say about this is that the changes we added to add the table to the FDS that are on top of the FF, are also few changes, I thought it could have been something I changed when I was putting this over FF.

That's why I'm thinking it seems like it's not related to these changes because they're in FF, so it seems like it's not related to that but I'm still worried about it because Markos rolled back the changes and it's back to working.

@markocikos, could you send a PR with the revert of the changes? Maybe we can go ahead with this to fix this issue as I look into it. Also because we will work on another ticket to add a new Table component to the FDS when we finish the new features we are adding.

markocikos commented on modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/FieldsSelectorDropdown.js:
```diff
@@ -57,7 +57,7 @@ const FieldsSelectorDropdown = ({fields}) => {
 					})}
 					displayType="secondary"
-					size="xs"
+					size={Liferay.FeatureFlags['LPS-193005'] ? 'xs' : undefined}
```
@matuzalemsteles This perfectly fixes part of the issue, that was a regression from your PR. First column size is unrelated to your change, I'll take a closer look and try to figure it out.

markocikos commented:
Followup in https://github.com/liferay-frontend/liferay-portal/pull/3887
